__tests__/workers/cdc/primary.ts

```diff
@@ -2412,3 +2425,230 @@ describe('comment downvote', () => {
     expect(triggerTypedEvent).toHaveBeenCalledTimes(0);
   });
 });
+
+describe('marketing cta', () => {
+  type ObjectType = MarketingCta;
+  const base: ChangeObject<ObjectType> = {
+    campaignId: 'worlds-best-campaign',
+    variant: 'card',
+    status: MarketingCtaStatus.Active,
+    createdAt: 0,
+    flags: {
+      title: 'Join the best community in the world',
+      description: 'Join the best community in the world',
+      ctaUrl: 'http://localhost:5002',
+      ctaText: 'Join now',
+    },
+  };
+
+  beforeEach(async () => {
+    await ioRedisPool.execute((client) => client.flushall());
+    await saveFixtures(con, User, usersFixture);
+    await saveFixtures(con, MarketingCta, [
+      { ...base, createdAt: new Date('2024-03-13 12:00:00') },
+    ]);
+    await saveFixtures(
+      con,
+      UserMarketingCta,
+      usersFixture.map((user) => ({
+        marketingCtaId: 'worlds-best-campaign',
+        userId: user.id,
+        createdAt: new Date('2024-03-13 12:00:00'),
+      })),
+    );
+
+    usersFixture.forEach(async (user) => {
+      await setRedisObject(
+        generateStorageKey(
+          StorageTopic.Boot,
+          StorageKey.MarketingCta,
+          user.id as string,
+        ),
+        // Don't really care about the value in these tests
+        'not null',
+      );
+    });
+  });
+
+  describe('on update', () => {
+    it('should clear boot cache for users assigned to the campaign and have not interacted with it', async () => {
+      await con.getRepository(UserMarketingCta).update(
+        {
+          userId: usersFixture[0].id,
+        },
+        {
+          readAt: new Date('2024-03-13 13:00:00'),
+        },
+      );
+
+      expect(
+        await getRedisObject(
```

Comment:
Can we test that `deleteRedisKey` was called appropriate amount of times. Just to validate it does not change more keys aside those we check. Then we can laso make sure that on `c` it is called 0 times.

Reply:
Hmm, the tests expect redis storage to be changed, so if I mock it, the keys aren't deleted from redis?

Reply:
I could instead just call redis instead to get a list of redis keys

Reply:
I mocked it and just called redis inside the mocked function 😈

Reply:
Ended up just checking for amount of keys in redis